src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -354,7 +354,7 @@ private function parseDefinition($id, $service, $file, array $defaults)
                 }
 
                 if ('deprecated' === $key) {
-                    $alias->setDeprecated(true, $value);
+                    $alias->setDeprecated(true, true === $value ? null : $value);
```

Comment:
should be reverted I think - at least we don't do that for services, so why do it for aliases?

Reply:
As I said in the last conversation :

Yes indeed, but if we specify `deprecated: true` in the alias configuration, without this condition an error is thrown because we do not give a string containing `%alias_id%` but `true`...

Reply:
can you check if we support that for Definition also? because if yes, there is a bug to fix with them - and if not, there is a bug to fix with aliases :) A test case would help secure this.

Reply:
I think there is a bug fix to do.
If we give `deprecated: true` in the configuration it crash like the alias configuration does.

I can do the same test than for alias to solve that

Reply:
If we support `true`, we need to support `false` I suppose, not really meaningful to me.
It's not [in the doc](https://symfony.com/doc/current/service_container/alias_private.html#deprecating-services) but [in the blog post](https://symfony.com/blog/new-in-symfony-4-3-deprecating-service-aliases) - which is wrong and should be fixed. The [article in 2015](https://symfony.com/blog/new-in-symfony-2-8-deprecated-service-definitions) was correct.
Let's just remove that change and done, it already fails you told me.

Reply:
blog post updated.

Reply:
so we doesn't support true and I remove all that thing ? (test included)

Reply:
Yep, `true` would be a new feature, thus unrelated - and unneeded to me.
If we can keep a test case ensuring there is a proper failure when `true` is passed, that'd be nice.

Reply:
ok, i'll do the change

Reply:
done